### ISSUE TITLE
BertForSequenceClassification finetune training loss and accuracy have some problem

### ISSUE THREAD

**Issue by good74152** [huggingface/transformers#9087]
## Environment info

     
- `transformers` version: v4.0.0
- Platform: colab pro
- Python version: 3.6.9
- PyTorch version (GPU?): 1.7.0+cu101
- Tensorflow version (GPU?): 2.3.0
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

@sgugger
@JetRunner

## Information
I follow the paper https://arxiv.org/pdf/2003.02245.pdf to do augmentation and test the performance

Model I am using Bert for BertTokenizer, BertForMaskedLM, BertForSequenceClassification

The problem arises when using:
Use Trainer to fine-tuning on both training set and concatenated set of training set and augmentation set, the Training log is No log or 0.683592, and accuracy is always 0.8

The tasks I am working on is:
An official GLUE  task: sst2, using by huggingface datasets package

The details:
Trainer setting I follow the examples/text_classification.ipynb to build the compute_metrics function and tokenize mapping function, but the training loss and accuracy have bug

my tokenized datasets format:
![擷取](https://user-images.githubusercontent.com/39672039/102020487-7d44ca80-3db4-11eb-8f48-590f885f6957.JPG)

compute_function, little modify by examples/text_classification.ipynb
![image](https://user-images.githubusercontent.com/39672039/102020602-24c1fd00-3db5-11eb-9ed3-5c3a55e50ac9.png)

bert_finetuned_setting
![image](https://user-images.githubusercontent.com/39672039/102020585-05c36b00-3db5-11eb-9440-aa715b3f4c36.png)

fine_tuned result
![image](https://user-images.githubusercontent.com/39672039/102020616-3c00ea80-3db5-11eb-9bcd-95380c8166fc.png)

**Comment by LysandreJik** [comment]
Hello, thanks for opening an issue! We try to keep the github issues for bugs/feature requests, rather than help with training.
Could you ask your question on the [forum](https://discusss.huggingface.co) instead?

Thanks!
